File: app/src/main/java/com/cn29/aac/ui/masterdetail/vm/SimpleMasterDetailShareViewModelFactory.java
package com.cn29.aac.ui.masterdetail.vm;

import com.cn29.aac.repo.github.GitRepoRepository;

import javax.inject.Inject;

import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

/**
 * Created by Charles Ng on 9/10/2017.
 */

public class SimpleMasterDetailShareViewModelFactory extends ViewModelProvider.NewInstanceFactory {

  GitRepoRepository gitRepoRepository;

  @Inject
  public SimpleMasterDetailShareViewModelFactory(GitRepoRepository gitRepoRepository) {
    this.gitRepoRepository = gitRepoRepository;
  }

  @Override
  public <T extends ViewModel> T create(Class<T> modelClass) {
    return (T) new SimpleMasterDetailShareViewModel(gitRepoRepository);
  }
}

File: app/src/main/java/com/cn29/aac/ui/masterdetail/vm/SimpleMasterDetailShareViewModelFactory.kt
package com.cn29.aac.ui.masterdetail.vm

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider.NewInstanceFactory
import com.cn29.aac.repo.github.GitRepoRepository
import javax.inject.Inject

/**
 * Created by Charles Ng on 9/10/2017.
 */
class SimpleMasterDetailShareViewModelFactory @Inject constructor(var gitRepoRepository: GitRepoRepository) :
        NewInstanceFactory() {
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return SimpleMasterDetailShareViewModel(gitRepoRepository) as T
    }

}